src/Microsoft.ML.Transforms/TensorFlow/Tensor.cs

```diff
@@ -0,0 +1,919 @@
+		}
+
+		/// <summary>
+		/// Creates a constant tensor with a single dimension from an ushort value.
+		/// </summary>
+		public unsafe TFTensor (ushort value)
+		{
+			var v = (ushort*)Marshal.AllocHGlobal (sizeof (ushort));
+			*v = value;
+			handle = TF_NewTensor (TFDataType.Int16, zeroDims: IntPtr.Zero, num_dims: 0, data: (IntPtr)v, len: (UIntPtr)sizeof (ushort), deallocator: FreeTensorDataDelegate, deallocator_arg: IntPtr.Zero);
+		}
+
+		/// <summary>
+		/// Creates a constant tensor with a single dimension from an byte value.
+		/// </summary>
+		public unsafe TFTensor (byte value)
+		{
+			var v = (int*)Marshal.AllocHGlobal (sizeof (byte));
+			*v = value;
+			handle = TF_NewTensor (TFDataType.UInt8, zeroDims: IntPtr.Zero, num_dims: 0, data: (IntPtr)v, len: (UIntPtr)sizeof (byte), deallocator: FreeTensorDataDelegate, deallocator_arg: IntPtr.Zero);
+		}
+
+		/// <summary>
+		/// Creates a constant tensor with a single dimension from a Complex value.
+		/// </summary>
+		public unsafe TFTensor (Complex value)
+		{
+			var v = (Complex*)Marshal.AllocHGlobal (sizeof (Complex));
+			*v = value;
+			handle = TF_NewTensor (TFDataType.Complex128, zeroDims: IntPtr.Zero, num_dims: 0, data: (IntPtr)v, len: (UIntPtr)sizeof (Complex), deallocator: FreeTensorDataDelegate, deallocator_arg: IntPtr.Zero);
+		}
+
+		/// <summary>
+		/// Creates a constant tensor with a single dimension from a float value.
+		/// </summary>
+		public unsafe TFTensor (float value)
+		{
+			var v = (float*)Marshal.AllocHGlobal (sizeof (float));
+			*v = value;
+			handle = TF_NewTensor (TFDataType.Float, zeroDims: IntPtr.Zero, num_dims: 0, data: (IntPtr)v, len: (UIntPtr)sizeof (float), deallocator: FreeTensorDataDelegate, deallocator_arg: IntPtr.Zero);
+		}
+
+		/// <summary>
+		/// Creates a constant tensor with a single dimension from a double value.
+		/// </summary>
+		public unsafe TFTensor (double value)
+		{
+			var v = (double*)Marshal.AllocHGlobal (sizeof (double));
+			*v = value;
+			handle = TF_NewTensor (TFDataType.Double, zeroDims: IntPtr.Zero, num_dims: 0, data: (IntPtr)v, len: (UIntPtr)sizeof (double), deallocator: FreeTensorDataDelegate, deallocator_arg: IntPtr.Zero);
+		}
+
+		/// <summary>
+		/// Creates a constant tensor with a single dimension from a long value.
+		/// </summary>
+		public unsafe TFTensor (long value)
+		{
+			var v = (long*)Marshal.AllocHGlobal (sizeof (long));
+			*v = value;
+			handle = TF_NewTensor (TFDataType.Int64, zeroDims: IntPtr.Zero, num_dims: 0, data: (IntPtr)v, len: (UIntPtr)sizeof (long), deallocator: FreeTensorDataDelegate, deallocator_arg: IntPtr.Zero);
+		}
+		/// <summary>
+		/// Creates a 1 dimensional tensor from an array of booleans.
+		/// </summary>
+		/// <param name="data">Data.</param>
+		public TFTensor (bool [] data) : base (SetupTensor (TFDataType.Bool, data, size: 1)) { }
```

Comment:
>SetupTensor [](start = 41, length = 11)

I think I need this method exposed so that I can use it in the transform.

Reply:
It's effectively exposed now that I have moved this code into the same assembly and its internal.  I need to understand the use case to determine if calling directly is the right thing to do.

---
In reply to: [210351552](https://github.com/zeahmed/machinelearning/pull/4#discussion_r210351552) [](ancestors = 210351552)